frontend/ui/widget/inputdialog.lua

```diff
@@ -724,8 +724,73 @@ function InputDialog:_addScrollButtons(nav_bar)
                 end,
             })
         end
-        -- Add a button to go to the line by its number in the file
         if self.fullscreen then
+            -- Add a button to search for a string in the edited text
+            table.insert(row, {
+                text = _("Find"),
+                callback = function()
+                    local input_dialog
+                    input_dialog = InputDialog:new{
+                        title = _("Enter text to search for"),
+                        stop_events_propagation = true, -- avoid interactions with upper InputDialog
+                        input = self.search_value,
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(input_dialog)
+                                    end,
+                                },
+                                {
+                                    text = _("Find first"),
+                                    callback = function()
+                                        self.search_value = input_dialog:getInputText()
```

Comment:
Does this saving of the input text survive a close/re-open of the text editor?
You store it in `self`, the instance at the moment, so I guess not.
In case you feel it's best to be able to get it back later, but it's not worth the pain of saving it in the settings, storing it in the `InputDialog` class object could give it a larger lifetime:
`InputDialog._last_search_text = input_dialog:getInputText()`

Reply:
Can you please explain a little bit: when will this suggested value be cleaned? When invoking another InputDialog, or freeing the memory for another module or something else? Is this leading `_` in the name significant?

Reply:
> when will this suggested value be cleaned? 

When you exit KOReader, or when re-doing a search and then replacing the input text.
If would be there when creating another new InputDialog.

Lesson of the day:
When you do `local InputDialog = require(".../inputdialog.lua")`, you get the single and unique cached `InputDialog` object in KOReader (what I call a class object).
When you do `local input_dialog = InputDialog:new({...})`, you create a clone, a new object, an _instance_ of the class `InputDialog`. From inside this instance, in its functions or in the callbacks, you use `self` to refer to this instance. You add/modify stuff into `self`, this doesn't pollute the class and other instances. This instance disappears when you're done with it, when you close the widget - and so does `self.search_value` . This is just right for most things.
But if you wanted to keep something shared by all instances, you could store the thing directly in the class object, using `InputDialog.the_thing = something_important` instead of `self.the_thing = something_temporary`. (This is done rarely, but in case you feel it's better for usability, you could use that trick.)

And no, the leading _ has no real significance, just a convention among developpers to say it's something a bit "private", internal. Otherwise, without `_`, other developpers could think it's a public option they could provide and override in `InputDialog:new({ search_value = "type something"})`.
We don't do that in all modules, but when I think about it, I try to do that, i.e.:
https://github.com/koreader/koreader/blob/ba00f3879dc970fb2d52091ab7a0a727f29ec6af/frontend/ui/widget/inputdialog.lua#L190-L197

Reply:
Thank you very much, I'll take note of that.
Rght here, I would keep as is, new file - new search.